motionplan/check.go

```diff
@@ -189,30 +147,89 @@ func checkPlanRelative(
 	if sfPlanner.planOpts, err = sfPlanner.plannerSetupFromMoveRequest(
 		planStartPoseWorld.Pose(),
 		planEndPoseWorld.Pose(),
-		startingInputs,
+		plan.Trajectory()[0],
 		worldState,
 		nil,
 		nil, // no pb.Constraints
 		nil, // no plannOpts
 	); err != nil {
 		return err
 	}
+	// change from 60mm to 30mm so we have finer interpolation along segments
+	sfPlanner.planOpts.Resolution = 30
 
-	// create a list of segments to iterate through
-	segments := make([]*ik.Segment, 0, len(plan.Path())-wayPointIdx)
+	currentInputs := executionState.CurrentInputs()
+	wayPointIdx := executionState.Index()
+	sf := sfPlanner.frame
+
+	// construct first segmenet's start configuration
 	// get checkFrame's currentInputs
 	// *currently* it is guaranteed that a relative frame will constitute 100% of a solver frame's dof
 	checkFrameCurrentInputs, err := sf.mapToSlice(currentInputs)
 	if err != nil {
 		return err
 	}
-	arcEndInputs, err := sf.mapToSlice(plan.Trajectory()[wayPointIdx])
+
+	// construct first segmenet's end configuration
+	currentWayPointTraj := plan.Trajectory()[wayPointIdx]
+	arcEndInputs, err := sf.mapToSlice(currentWayPointTraj)
+	if err != nil {
+		return err
+	}
+
+	// construct first segmenet's startPosition
+	currentPoses := executionState.CurrentPoses()
```

Comment:
> A lot of this seems like it should be unnecessary now

Since slamLocalizer, movementSensorLocalizer, and yForwards2dLocalizer all return PoseInFrame for a CurrentPosition call with Parent set to referenceframe.World I agree that calling `toWorld` seems unnecessary here.

> can we replace most of this (and all the toWorld calls) with fss.Transform calls?

I am confused by this since `toWorld` already has `fs.Transform(inputs, pif, referenceframe.World)` inside its definition.
On a different note, `toWorld` in this PR is called exactly four times.
In all these cases, the poseInFrame we are transforming into world already has its parent == referenceframe.World. Hence it doesn't really make sense for `toWorld` or `validateRelPiF` to continue exisiting. I will remove them.

Reply:
In fact, since we no longer need `toWorld` we also no longer need `fs referenceframe.FrameSystem` as an argument for `checkPlanRelative`.